### PR TITLE
Add error code support

### PR DESCRIPTION
**Pull request by hangrymuppet**

### CODE REVIEW

File: src/treemodel.cpp [hunk 1]
```diff
@@ -617,26 +636,30 @@ QString TreeModel::JsonToString(const QJsonValue& json, const bool isSingleLine)
...
+    
+    if (hasART || hasErrorCode) {
+        QJsonObject obj;
+
+        if (eventObject.contains("v") && eventObject["v"].type() == QJsonValue::Object)
+            obj = eventObject["v"].toObject();
+        else
+            obj["v"]=eventObject["v"]; // Create new object with "v"
+
+        if (eventObject.contains("a")) {
```
Comment (terencesoh) [inline src/treemodel.cpp]: Could use hasART.

File: src/treemodel.cpp [hunk 2]
```diff
@@ -617,26 +636,30 @@ QString TreeModel::JsonToString(const QJsonValue& json, const bool isSingleLine)
...
         }
-        else
-        {
-            // "v" is not an object, create a new object and put "v" and "a" as children
-            QJsonObject obj;
-            obj["v"]=eventObject["v"];
-            obj["~art"]=eventObject["a"];
-            return std::move(obj);
+
+        if (eventObject.contains("e")) {
```
Comment (terencesoh) [inline src/treemodel.cpp]: Could use hasErrorCode.

File: src/treemodel.cpp [hunk 3]
```diff
@@ -455,11 +468,17 @@ void TreeModel::SetupChild(TreeItem *child, const QJsonObject & event)
...
     bool hasArtData = event.contains("a");
+    bool hasErrorCode = event.contains("e");
     if (hasArtData)
     {
         // 0xE2978F = BLACK CIRCLE
         child->SetData(COL::ART, QString::fromUtf8("\xE2\x97\x8F"));
     }
+    if (hasErrorCode)
+    {
+        // 0xE2978F = BLACK CIRCLE
```
Comment (terencesoh) [inline src/treemodel.cpp]: Perhaps a different symbol to make it easier to differentiate ART and error code.

File: src/treemodel.cpp (hunk 1 above)
Comment (bajajh) [reply src/treemodel.cpp]: Doh! fixed

File: src/treemodel.cpp (hunk 2 above)
Comment (bajajh) [reply src/treemodel.cpp]: Doh again! Fixed!

File: src/treemodel.cpp (hunk 3 above)
Comment (bajajh) [reply src/treemodel.cpp]: Changed to black square.

**Pull request merged**